Runtime/Services/BacktraceSession.cs

```diff
@@ -0,0 +1,354 @@
+            {
+                Debug.LogWarning("Attribute name is not available in attribute scope. Please define attribute to set unique event.");
+                return false;
+            }
+            var @event = new UniqueEvent(attributeName, DateTimeHelper.Timestamp(), attributes);
+            UniqueEvents.AddLast(@event);
+            return true;
+        }
+
+        /// <summary>
+        /// Determine if Backtrace Session can add next event to store
+        /// </summary>
+        /// <param name="name">event name</param>
+        /// <returns>True if we're able to add. Otherwise false.</returns>
+        private bool ShouldProcessEvent(string name)
+        {
+            return !string.IsNullOrEmpty(name) && (_maximumNumberOfEventsInStore == 0 || (Count() + 1 <= _maximumNumberOfEventsInStore));
+        }
+
+
+        /// <summary>
+        /// Get number of events available right now in store
+        /// </summary>
+        /// <returns>number of events in store</returns>
+        public int Count()
+        {
+            return UniqueEvents.Count + SessionEvents.Count;
+        }
+
+        /// <summary>
+        /// Add session event to next Backtrace session request
+        /// </summary>
+        /// <param name="eventName">session event name</param>
+        public bool AddSessionEvent(string eventName, IDictionary<string, string> attributes = null)
+        {
+            if (!ShouldProcessEvent(eventName))
+            {
+                Debug.LogWarning("Skipping report: Reached store limit or event has empty name.");
+                return false;
+            }
+
+            var @event = new SessionEvent(eventName, DateTimeHelper.Timestamp(), attributes);
+            SessionEvents.AddLast(@event);
+            return true;
+        }
+
+        /// <summary>
+        /// Clean unique and session events stored in Backtrace Session
+        /// </summary>
+        public void ClearEvents()
+        {
+            UniqueEvents.Clear();
+            SessionEvents.Clear();
+        }
+
+        private void OnRequestCompleted()
+        {
+            _numberOfDroppedRequests = 0;
+            return;
+        }
+
+        private BacktraceJObject CreateJsonPayload(ICollection<UniqueEvent> uniqueEvents, ICollection<SessionEvent> sessionEvents)
+        {
+            var jsonData = new BacktraceJObject();
+            jsonData.Add("application", Application.productName);
+            jsonData.Add("appversion", Application.version);
+            jsonData.Add("metadata", CreatePayloadMetadata());
+
+            // add unique events
+            var uniqueEventsJson = new List<BacktraceJObject>();
+            foreach (var uniqueEvent in uniqueEvents)
+            {
+                uniqueEventsJson.Add(uniqueEvent.ToJson());
+            }
+
+            jsonData.Add("unique_events", uniqueEventsJson);
+
+            // add unique events

```

Comment:
Typo in comment, should be `add session events`

Reply:
Thanks!